wear/src/main/java/io/homeassistant/companion/android/home/views/ChooseEntityView.kt

```diff
@@ -0,0 +1,99 @@
+import androidx.wear.compose.material.rememberScalingLazyListState
+import com.mikepenz.iconics.compose.Image
+import com.mikepenz.iconics.typeface.library.community.material.CommunityMaterial
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.util.RotaryEventState
+import io.homeassistant.companion.android.util.SetTitle
+import io.homeassistant.companion.android.util.getIcon
+
+@Composable
+fun ChooseEntityView(
+    validEntities: List<Entity<*>>,
+    onNoneClicked: () -> Unit,
+    onEntitySelected: (entityId: String) -> Unit
+) {
+    val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+    RotaryEventState(scrollState = scalingLazyListState)
+    ScalingLazyColumn(
+        modifier = Modifier
+            .fillMaxSize(),
+        contentPadding = PaddingValues(
+            top = 40.dp,
+            start = 8.dp,
+            end = 8.dp,
+            bottom = 40.dp
+        ),
+        verticalArrangement = Arrangement.spacedBy(4.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        state = scalingLazyListState
+    ) {
+        item {
+            SetTitle(id = R.string.shortcuts)
+        }
+        item {
+            Chip(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(bottom = 16.dp),
+                icon = { Image(asset = CommunityMaterial.Icon.cmd_delete) },
+                label = { Text(text = "None") },
+                onClick = onNoneClicked,
+                colors = ChipDefaults.primaryChipColors(
+                    contentColor = Color.Black
+                )
+            )
+        }
+        items(validEntities.size) { index ->
+            val attributes = validEntities[index].attributes as Map<*, *>
+            val iconBitmap = getIcon(
+                attributes["icon"] as String?,
+                validEntities[index].entityId.split(".")[0],
+                LocalContext.current
+            )
+            Chip(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                icon = {
+                    Image(
+                        asset = iconBitmap ?: CommunityMaterial.Icon.cmd_cellphone,
+                        colorFilter = ColorFilter.tint(Color.White)
+                    )
+                },
+                label = {
+                    Text(
+                        text = attributes["friendly_name"].toString(),
+                        maxLines = 2,
+                        overflow = TextOverflow.Ellipsis
+                    )
+                },
+                enabled = validEntities[index].state != "unavailable",
+                onClick = {
+                    val elementString = "${validEntities[index].entityId},${attributes["friendly_name"]},${attributes["icon"]}"
+                    onEntitySelected(elementString)
```

Comment:
This should pass that new object that we are using rather than a concatenated string.